### DISCUSSION TITLE
FlowProducer and when to use them?

### DISCUSSION THREAD

**Discussion by doebi** [taskforcesh/bullmq#1517] [Q&A]
I am trying to find an adequate solution to implementing following process in BullMQ.

```
* acquire information A
* process information from A to B
  - publish B to message broker
  - store B in DBMS 1
* process information from B to C
  - publish C to message broker
  - store C in DBMS 2
```

I read about FlowProducer and initally thought they might be a good fit. But from what I saw they are good at having multiple children jobs aggregated in one parent job which processes their output. So in my case If I map all the steps in a flow I would need multiple parent jobs, which is not possible.

My next idea is to create a Flow with steps [acquire, process, process] and have them add jobs to separate queues for DBMS 1, DBMS 2 and message broker. But in that case, do I still need the FlowProducer?

What are your thoughts?

**Comment by roggervalf** [comment]
hi @doebi sorry for the delay, but did you also check our pattern for step jobs? https://docs.bullmq.io/patterns/process-step-jobs
